src/Breakpoint.php

```diff
@@ -2,80 +2,104 @@
 
 namespace Spatie\ResponsiveImages;
 
+use Exception;
 use Illuminate\Contracts\Support\Arrayable;
 use Illuminate\Support\Collection;
 use Illuminate\Support\Facades\Config;
 use League\Flysystem\FilesystemException;
-use Spatie\ResponsiveImages\Jobs\GenerateImageJob;
 use Statamic\Contracts\Assets\Asset;
 use Statamic\Facades\Blink;
 use Statamic\Facades\Glide as GlideManager;
 use Statamic\Imaging\ImageGenerator;
 use Statamic\Support\Str;
 
+/**
+ * @property-read \Statamic\Assets\Asset $asset
+ * @property-read string $label
+ * @property-read int $minWidth
+ * @property-read array $params
+ * @property-read string $widthUnit
+ */
 class Breakpoint implements Arrayable
 {
-    /** @var \Statamic\Assets\Asset */
-    public $asset;
+    /** @var Asset */
+    public Asset $asset;
 
     /** @var string */
-    public $label;
+    public string $label;
 
-    /** @var ?int */
-    public $value;
+    /**
+     * @var int The minimum width of when the breakpoint starts
+     */
+    public int $minWidth;
 
     /** @var array */
-    public $parameters;
-
-    /** @var float */
-    public $ratio;
+    public array $params;
 
     /** @var string */
-    public $unit;
+    public string $widthUnit;
+
+    /** @var Collection<Source> */
+    private Collection $sources;
 
-    public function __construct(Asset $asset, string $label, int $value, array $parameters)
+    public function __construct(Asset $asset, string $label, int $breakpointMinWidth, array $breakpointParams)
     {
         $this->asset = $asset;
         $this->label = $label;
-        $this->value = $value;
-        $this->parameters = $parameters;
-        $this->ratio = $this->parameters['ratio'] ?? $this->asset->width() / $this->asset->height();
+        $this->minWidth = $breakpointMinWidth;
+        $this->params = $breakpointParams;
+        $this->widthUnit = config('statamic.responsive-images.breakpoint_unit', 'px');
+    }
 
-        $this->unit = config('statamic.responsive-images.breakpoint_unit', 'px');
+    public function __set($name, $value): void
+    {
+        throw new Exception(sprintf('Cannot modify property %s', $name));
```

Comment:
Read only properties in classes are available only starting with PHP 8.1. As we are exposing `Breakpoint` and `Source` classes now for DimensionCalculator, as such need to be aware what methods and properties are available.